swarm/storage/resource_test.go

```diff
@@ -65,161 +70,105 @@ func (r *FakeRPC) BlockNumber() (string, error) {
 
+	// check that the metadata retrieved from the chunk matches what we gave it
 	if recoveredaddress != originaladdress {
 		teardownTest(t, fmt.Errorf("addresses dont match: %x != %x", originaladdress, recoveredaddress))
 	}
-	teardownTest(t, nil)
-}
-
-// determine resource update metadata from chunk data
-func TestResourceReverseLookup(t *testing.T) {
 
-	// privkey for signing updates
-	privkey, err := crypto.GenerateKey()
-	if err != nil {
-		return
+	if !bytes.Equal(key[:], chunk.Key[:]) {
+		teardownTest(t, fmt.Errorf("Expected chunk key '%x', was '%x'", key, chunk.Key))
 	}
-
-	// make fake backend, set up rpc and create resourcehandler
-	backend := &fakeBackend{
-		blocknumber: startBlock,
+	if period != checkperiod {
+		teardownTest(t, fmt.Errorf("Expected period '%d', was '%d'", period, checkperiod))
 	}
-	rh, _, err, teardownTest := setupTest(privkey, backend, nil)
-	if err != nil {
-		teardownTest(t, err)
+	if version != checkversion {
+		teardownTest(t, fmt.Errorf("Expected version '%d', was '%d'", version, checkversion))
 	}
-
-	// create a new resource
-	rsrc, err := rh.NewResource(domainName, resourceFrequency)
-	if err != nil {
-		teardownTest(t, err)
-	}
-
-	// update data
-	fwdBlocks(int(resourceFrequency+1), backend)
-	data := []byte("foo")
-	resourcekey, err := rh.Update(domainName, data)
-	if err != nil {
-		teardownTest(t, err)
-	}
-	chunk, err := rh.ChunkStore.(*resourceChunkStore).localStore.(*LocalStore).memStore.Get(Key(resourcekey))
-	if err != nil {
-		teardownTest(t, err)
+	if safeName != checkname {
+		teardownTest(t, fmt.Errorf("Expected name '%s', was '%s'", safeName, checkname))
 	}
-
-	// check if data after header length offset is as expected
-	headerlength := binary.LittleEndian.Uint16(chunk.SData[signatureLength : signatureLength+2])
-	if !bytes.Equal(chunk.SData[signatureLength+headerlength+2:], data) {
-		teardownTest(t, fmt.Errorf("Expected chunk data with header length %d (pos %d) to match %x, but was %x", headerlength, signatureLength+headerlength+2, data, chunk.SData[signatureLength+headerlength+2:]))
-	}
-
-	// get name, period, version from chunk and check
-	revperiod, revversion, revname, revdata, err := parseUpdate(chunk.SData[signatureLength:])
-
-	if !bytes.Equal(revname, rsrc.nameHash.Bytes()) {
-		teardownTest(t, fmt.Errorf("Expected retrieved name from chunk data to be '%x', was '%x'", rsrc.nameHash.Bytes(), revname))
-	}
-	if !bytes.Equal(revdata, data) {
-		teardownTest(t, fmt.Errorf("Expected retrieved data from chunk data to be '%x', was '%x'", data, revdata))
-	}
-
-	if revperiod != 2 {
-		teardownTest(t, fmt.Errorf("Expected retrieved period from chunk data to be 1, was %d", revperiod))
-	}
-	if revversion != 1 {
-		teardownTest(t, fmt.Errorf("Expected retrieved version from chunk data to be 1, was %d", revversion))
+	if !bytes.Equal(data, checkdata) {
+		teardownTest(t, fmt.Errorf("Expectedn data '%x', was '%x'", data, checkdata))
```

Comment:
It feels a little awkward to have to call `teardownTest` rather than the typical `t.Fatal`, typically tests just teardown in a `defer`, something like:

```go
teardownTest := setupTest(t, ...)
defer teardownTest()
...
if err != nil {
	t.Fatal(err)
}
```